src/main/kotlin/com/demonwav/mcdev/platform/mcp/actions/CopyATAction.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mcp.actions
+
+import com.demonwav.mcdev.platform.mcp.McpModuleType
+import com.demonwav.mcdev.platform.mixin.util.findFirstShadowTarget
+import com.demonwav.mcdev.util.getDataFromActionEvent
+import com.demonwav.mcdev.util.invokeLater
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataKeys
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.ui.popup.Balloon
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.wm.WindowManager
+import com.intellij.psi.*
+import com.intellij.ui.LightColors
+import com.intellij.ui.awt.RelativePoint
+import java.awt.Toolkit
+import java.awt.datatransfer.StringSelection
+
+class CopyATAction : AnAction() {
+    override fun actionPerformed(e: AnActionEvent) {
+        val data = getDataFromActionEvent(e) ?: return showBalloon("Unknown failure", e)
+
+        if (data.element !is PsiIdentifier) {
+            showBalloon("Not a valid element", e)
+            return
+        }
+
+        val mcpModule = data.instance.getModuleOfType(McpModuleType) ?: return showBalloon("No mappings found", e)
+
+        mcpModule.srgManager?.srgMap?.done { srgMap ->
```

Comment:
Can we avoid duplicating this part from `FindSrgMappingAction`? It should be possible to move this to a separate method since the code appears to be almost identical.

Reply:
Certainly. What would be the best way to do this in Kotlin, my first assumption would be to make them have a common superclass which includes to common functions?

Reply:
I haven't looked into the exact details regarding your implementation, unfortunately due to lack of time. But depending on the context of the situation, either a common super class, or top level functions would work well. Intuition regarding what you're working on tells me a commons super class is likely the cleanest solution.

If you want top level functions instead, and you want to namespace the functions a bit you can place them in an `object`. Unfortunately Kotlin doesn't have `package-private` visibility to help clean up the global namespace, so sometimes `object`s are the best way to keep it clean